### ISSUE TITLE
Upgrading to 0.0.12 breaks exporter-zipkin causes application to crash

### ISSUE THREAD

**Issue by drubin** [census-instrumentation/opencensus-node#526]
### What version of OpenCensus are you using?
0.0.12

### What version of Node are you using?
v10.15.3

### What did you do?
Running the following code breaks with 0.0.12 but works with 0.0.11. 

In order to trigger the crash, you need to  generate some spans and wait < 1 minute it should cause a crash

```js
const port = process.env.PORT || 3000;
const upstream_uri = process.env.UPSTREAM_URI || 'http://time.jsontest.com/';
const service_name = process.env.SERVICE_NAME || 'opencensus-test-1-v1';

const tracing = require('@opencensus/nodejs');
const propagation = require('@opencensus/propagation-b3');
const b3 = new propagation.B3Format();
const zipkin = require('@opencensus/exporter-zipkin');

const exporter = new zipkin.ZipkinTraceExporter({
  url: 'http://localhost:9411/api/v2/spans',
  serviceName: 'test',
  debug: true,
  logLevel: 10
})

tracing.start({
  propagation: b3,
  samplingRate: 1.0,
  bufferSize: 1,
  bufferTimeout: 1,
  exporter
})

const express = require('express')
const app = express()
const request = require('request-promise-native')

app.get('/', async(req, res) => {
  await request({url: upstream_uri})
  res.end(`done`)
})

app.listen(port, () => {
  console.log(`${service_name} listening on port ${port}!`)
})
```
After some time I get the following 

```
events.js:174
      throw er; // Unhandled 'error' event
      ^

Error: socket hang up
    at createHangUpError (_http_client.js:323:15)
    at Socket.socketOnEnd (_http_client.js:426:23)
    at Socket.emit (events.js:194:15)
    at endReadableNT (_stream_readable.js:1125:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
Emitted 'error' event at:
    at Socket.socketOnEnd (_http_client.js:426:9)
    at Socket.emit (events.js:194:15)
    at endReadableNT (_stream_readable.js:1125:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

My package.json

```
    "@opencensus/exporter-jaeger": "0.0.12",
    "@opencensus/exporter-zipkin": "0.0.12",
    "@opencensus/nodejs": "0.0.12",
    "@opencensus/propagation-b3": "0.0.12",
```

### What did you expect to see?
Sending spans to zipkin

### What did you see instead?
Not only did the tracing data not get to zipkin but it causes the application to crash with an exit code of 1

Error 

```
events.js:174
      throw er; // Unhandled 'error' event
      ^

Error: socket hang up
    at createHangUpError (_http_client.js:323:15)
    at Socket.socketOnEnd (_http_client.js:426:23)
    at Socket.emit (events.js:194:15)
    at endReadableNT (_stream_readable.js:1125:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
Emitted 'error' event at:
    at Socket.socketOnEnd (_http_client.js:426:9)
    at Socket.emit (events.js:194:15)
    at endReadableNT (_stream_readable.js:1125:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

### Additional context
It works on 0.0.11 and also works with jeager exporter on 0.0.12

**Comment by mayurkale22** [comment]
@drubin thanks for filling the issue, I am looking into it.

**Comment by drubin** [comment]
Let me know if i can help in any way. Still new to this but I can try help

**Comment by mayurkale22** [comment]
> Let me know if i can help in any way. Still new to this but I can try help

I think I found the issue, just checking why it's working in Jaeger case. I will update thread shortly.

**Comment by mayurkale22** [comment]
@drubin  The "**Socket hang up**" error in Zipkin exporter's http request caused by infinite loop while getting `SpanContext` (get traceState()). I have fixed the issue in #530, these changes have all been tested locally against a real local zipkin instance and been confirmed to be working. 

We planned to cut a release(0.0.13) soon with the fix.

**Comment by drubin** [comment]
@mayurkale22 awesome let me know and I will gladly give it a go.

**Comment by jinmel** [comment]
@mayurkale22 FYI, I saw this error for stackdriver exporter. I'll take a look myself if it is related.

**Comment by mayurkale22** [comment]
@jinmel Thanks for the help, I have tested the changes (#530) locally against a real stackdriver trace exporter and been confirmed to be working. Please let me know if you find anything weird.

**Comment by mayurkale22** [comment]
@drubin FYI -> https://github.com/census-instrumentation/opencensus-node/pull/539

**Comment by mayurkale22** [comment]
The fix the available with the new release (`0.0.13`), I have verified the changes with given example and been confirmed to be working. Closing this issue now, let us know if you encounter any issues!
